src/main/java/org/spongepowered/api/world/BlockChangeFlag.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.world;
+
+import com.google.common.base.Objects;
+
+/**
+ * A flag of sorts that determines whether a block change will perform various
+ * interactions, such as notifying neighboring blocks, performing block phsyics
+ * on placement, etc.
+ */
+public enum BlockChangeFlag {
+
+    ALL(Flags.NEIGHBOR_MASK | Flags.PHYSICS_MASK),
+    NEIGHBOR(Flags.NEIGHBOR_MASK),
+    PHYSICS(Flags.PHYSICS_MASK),
+    NEIGHBOR_PHYSICS(Flags.NEIGHBOR_MASK | Flags.PHYSICS_MASK),
```

Comment:
That really doesn't matter since when a new flag is added, it'll still work the exact same way, except `ALL` won't be matched because of the obvious bit masked flags.